drivers/display/display_st7528i.c

```diff
@@ -0,0 +1,513 @@
+/*
+ * Copyright (c) 2022-2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT sitronix_st7528i
+
+#include "display_st7528.h"
+
+#include <errno.h>
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(display_st7528i, CONFIG_DISPLAY_LOG_LEVEL);
+
+/* All following bytes will contain commands */
+#define ST7528I_CONTROL_ALL_BYTES_CMD  0x00
+/* All following bytes will contain data */
+#define ST7528I_CONTROL_ALL_BYTES_DATA 0x40
+/* The next byte will contain a command */
+#define ST7528I_CONTROL_ONE_BYTE_CMD   0x80
+/* The next byte will contain data */
+#define ST7528I_CONTROL_ONE_BYTE_DATA  0xC0
+
+struct st7528i_config {
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec reset;
+	struct gpio_dt_spec chip_select;
+
+	uint16_t height;
+	uint16_t width;
+
+	uint8_t framerate;
+	uint8_t com_partial_display;
+	uint8_t com_offset;
+	bool invert_com;
+	bool invert_segments;
+
+	uint8_t lcd_bias;
+	uint8_t regulator_resistor;
+	uint8_t electronic_volume;
+	uint8_t boost;
+};
+
+struct st7528i_data {
+	uint8_t framebuffer[ST7528_NUM_PAGES * ST7528_BPP * ST7528_WIDTH];
```

Comment:
Because setting the value of any one pixel also affects 7 neighbouring pixels. So either the driver has to keep track of the framebuffer itself, or perform a read of the controller framebuffer for every pixel that is set - since I2C reads seem to be broken (the datasheet kind of suggests that they aren't supported at all), the latter isn't even possible.

Reply:
>Because setting the value of any one pixel also affects 7 neighbouring pixels.

That is not necessarily a reason if the tile areas are arbitrarily addressable.